### PR TITLE
Performance best practices

### LINKED ISSUE #9526: Performance best practices
@mjrousos to provide first draft.

### PR DESCRIPTION
**Pull request by mjrousos**
Adds a document listing common performance pitfalls in ASP.NET Core apps and how to avoid them.

Fixes #9526 

@Rick-Anderson 	edit: [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/performance/performance-best-practices?view=aspnetcore-1.1&branch=pr-en-us-9590)

### CODE REVIEW

**Comment by tdykstra** [comment]
For now I would put this under the Other topics node of the TOC (see aspnetcore/toc.md). We can decide about a top-level Performance node and what belongs in it later.

**Comment by mjrousos** [comment]
I think it would be valuable to have a top-level Performance node (including sections like diagnostics tools, performance features, and best practices (this doc)). I'm fine with putting this in Other topics for the moment, though, until we have more content for the performance node.

Is there any reason not to have a perf node other than lack of content?

**Comment by tdykstra** [comment]
No reason.  We had it before the last TOC reorg but got rid of it because we didn't really have enough content to justify it.

File: aspnetcore/performance/performance-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+* **Do** use ASP.NET Core's [built-in support](https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification) for bundling and minifying client assets.
+* **Do** consider other third-party tools like [Gulp](https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification#consume-bundleconfigjson-from-gulp) or [Webpack](https://webpack.js.org/) for more complex client asset management.
+
+## Use the Latest ASP.NET Core Releases
+With every ASP.NET Core release, performance work is done. Optimtizations in .NET Core and additional ASP.NET Core performance features mean that newer versions of ASP.NET Core will outperform older versions. For example, .NET Core 2.1 addded support for compiled regular expressions and benefitted from [`Span<T>`](https://msdn.microsoft.com/en-us/magazine/mt814808.aspx). ASP.NET Core 2.2 will bring support for HTTP/2. If performance is a priority, it may worthwhile upgrading to a recent ASP.NET Core version and taking advantage of new [performance features](TBD).
```
Comment (conniey) [inline aspnetcore/performance/performance-best-practices.md]: spelling: addded -> added

**Comment by Rick-Anderson** [comment]
@tdykstra 	@scottaddie 	@guardrex 	hold feedback. I'm PR'ing this PR.

**Comment by Rick-Anderson** [comment]
@mjrousos 	I wasn't able to create a PR to merge into this PR, so I created #9645. I'm just getting started, it's not ready for review. Maybe @scottaddie 	can tell you how to create PR's so I can more easily PR your PR.

**Comment by tdykstra** [comment]
Closing this since #9645 added this doc.

**Pull request closed without merging**